IdentityCore/src/throttling/cache/MSIDThrottlingCacheService.h

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+@class MSIDThrottlingCacheRecord;
+
+//singleton instances of MSIDServerDelayCache & MSIDUIRequiredCache will be in the implementation file as private properties.
+//Additionally, there will be dummy head and tail node in the .m file to maintain pseudo doubly-linked-list so we can better handle all corner cases.
+@interface MSIDThrottlingCacheService : NSObject
+
+@property (nonatomic, readonly) NSUInteger cacheSizeUI;
+@property (nonatomic, readonly) NSUInteger cacheSizeServerDelay;
+
+- (instancetype)initThrottlingCacheService:(NSUInteger cacheSizeUI)
+                      cacheSizeServerDelay:(NSUInteger cacheSizeServerDelay);
+
+//add new node to the front of LRU cache.
+//if node already exists, update and move it to the front of LRU cache
+- (void)addRequestToUICache:(NSString *)thumbprintKey
+              errorResponse:(NSError *)errrorResponse;
```

Comment:
Not sure if it applies, is the LRU cache thread safe? Should we consider it?

Reply:
+1, Thread safe is super important for this cache module but Peter is focusing the thumbprint calculator for now, I think he will address it when he implements cache service

Reply:
@kaisong1990, yep. we will rely on MSIDCache for thread-safety :) As Hieu mentioned, it will become more clear once I start working on the cache service :) Thank you for calling it out.